restate-test-utils/src/test_utils.rs

```diff
@@ -0,0 +1,211 @@
+use serde::{Serialize, Deserialize};
+use restate_sdk::{discovery::Service, errors::HandlerError, prelude::{Endpoint, HttpServer}};
+use tokio::{io::{self, AsyncWriteExt}, task::{self, JoinHandle}};
+use std::time::Duration;
+
+// addapted from from restate-admin-rest-model crate version 1.1.6
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RegisterDeploymentRequestHttp {
+    uri: String,
+    additional_headers:Option<Vec<(String, String)>>,
+    use_http_11: bool,
+    force: bool,
+    dry_run: bool
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RegisterDeploymentRequestLambda {
+    arn: String,
+    assume_role_arn: Option<String>,
+    force: bool,
+    dry_run: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct VersionResponse {
+    version:String,
+    min_admin_api_version:u32,
+    max_admin_api_version:u32
+}
+
+pub struct TestContainer {
+    container:ContainerAsync<GenericImage>,
+    stdout_logging:JoinHandle<()>,
+    stderr_logging:JoinHandle<()>,
+    endpoint:Option<JoinHandle<()>>
+}
+
+
+impl TestContainer {
+
+    //"docker.io/restatedev/restate", "latest"
+    pub async fn new(image:&str, version:&str) -> Result<Self, HandlerError> {
+
+        let image = GenericImage::new(image, version)
+            .with_exposed_port(9070.tcp())
+            .with_exposed_port(8080.tcp())
+            .with_wait_for(WaitFor::message_on_stdout("Ingress HTTP listening"));
+
+        // have to expose entire host network because testcontainer-rs doesn't implement selective SSH port forward from host
+        // see https://github.com/testcontainers/testcontainers-rs/issues/535
+        let container = ContainerRequest::from(image)
+            .with_host("host.docker.internal" , testcontainers::core::Host::HostGateway)
+            .start()
+            .await?;
+
+        let mut container_stdout = container.stdout(true);
+        // Spawn a task to copy data from the AsyncBufRead to stdout
+        let stdout_logging = task::spawn(async move {
+            let mut stdout = io::stdout();
+            if let Err(e) = io::copy(&mut container_stdout, &mut stdout).await {
+                eprintln!("Error copying data: {}", e);
+            }
+        });
+
+        let mut container_stderr = container.stderr(true);
+        // Spawn a task to copy data from the AsyncBufRead to stderr
+        let stderr_logging = task::spawn(async move {
+            let mut stderr = io::stderr();
+            if let Err(e) = io::copy(&mut container_stderr, &mut stderr).await {
+                eprintln!("Error copying data: {}", e);
+            }
+        });
+
+        let host = container.get_host().await?;
+        let ports = container.ports().await?;
+
+        let admin_port = ports.map_to_host_port_ipv4(9070.tcp()).unwrap();
+
+        let admin_url = format!("http://{}:{}/version", host, admin_port);
+        reqwest::get(admin_url)
+            .await?
+            .json::<VersionResponse>()
+            .await?;
+
```

Comment:
Maybe add a log line here (with `tracing::info`) saying restate container started with the forwarded ports